develop/tutorials/articles/290-configuration/03-fully-custom-system-settings-ui.markdown

```diff
@@ -0,0 +1,399 @@
+        return "site-navigation-language-web-template-configuration-name";
+    }
+
+The `getName` method returns the localized value of the language key. It calls
+`LanguageUtil.get` to retrieve the localized value of the entry title from the
+resource bundle providing the module's localization keys and values. Pass in the
+locale (provided to the method by the code that calls it), the configuration
+screen's class name, and the String returned by `getKey` to retrieve the right
+resource bundle.
+
+    @Override
+    public String getName(Locale locale) {
+        return LanguageUtil.get(
+            ResourceBundleUtil.getBundle(
+                locale, LanguageTemplateConfigurationScreen.class),
+            getKey());
+    }
+
+Set the scope of the entry with the `getScope` method:
+
+    @Override
+    public String getScope() {
+        return "system";
+    }
+
+Provide the render logic via a `render` method. The rendering approach
+demonstrated here uses a JSP backed by a `DisplayContext` class that's set into
+the request object so the values set from this `render` method are available in
+the JSP via the `DisplayContext` object's getters. Here, loop through the DDM
+Template Keys for the given `groupId` and set them into the display context with
+the `addTemplateKey` method. Then set the other necessary values that the JSP
+will need. In this case, set the title, the field label, and the redirect URL.
+Lastly, call `renderJSP` and pass in the `servletContext`, request, response,
+and the path to the JSP: 
+
+    @Override
+    public void render(HttpServletRequest request, HttpServletResponse response)
+        throws IOException {
+
+        Locale locale = LocaleThreadLocal.getThemeDisplayLocale();
+
+        LanguageTemplateConfigurationDisplayContext
+            languageTemplateConfigurationDisplayContext =
+                new LanguageTemplateConfigurationDisplayContext();
+
+        languageTemplateConfigurationDisplayContext.setCurrentTemplateName(
+            _siteNavigationLanguageWebTemplateConfiguration.ddmTemplateKey());
+
+        long groupId = 0;
+
+        Group group = _groupLocalService.fetchCompanyGroup(
+            CompanyThreadLocal.getCompanyId());
+
+        if (group != null) {
+            groupId = group.getGroupId();
+        }
+
+        List<DDMTemplate> ddmTemplates = _ddmTemplateLocalService.getTemplates(
+            groupId, _portal.getClassNameId(LanguageEntry.class));
+
+        for (DDMTemplate ddmTemplate : ddmTemplates) {
+            languageTemplateConfigurationDisplayContext.addTemplateValue(
+                ddmTemplate.getTemplateKey(), ddmTemplate.getName(locale));
+        }
+
+        languageTemplateConfigurationDisplayContext.setRedirect(
+            _portal.getCurrentURL(request));
+
+        languageTemplateConfigurationDisplayContext.setTitle(getName(locale));
+
+        languageTemplateConfigurationDisplayContext.setFieldLabel(
+            "Language Selection Style");
```

Comment:
Whoops, forgot to make a language key for this.   I'd probably avoid doing a literal translation in our official documentation ;)  Sorry about that.